File: app/src/main/java/com/pixplicity/cryptogram/utils/EventProvider.java
package com.pixplicity.cryptogram.utils;

import android.os.Handler;
import android.os.Looper;

import com.squareup.otto.Bus;

public class EventProvider {

    private static EventProvider sInstance;

    private final Handler mHandler = new Handler(Looper.getMainLooper());
    private final Bus mBus = new Bus();

    public static EventProvider getInstance() {
        if (sInstance == null) {
            sInstance = new EventProvider();
        }
        return sInstance;
    }

    private EventProvider() {
    }

    public static Bus getBus() {
        return getInstance().mBus;
    }

    public static void postEvent(final Object event) {
        if (Looper.myLooper() == Looper.getMainLooper()) {
            getBus().post(event);
        } else {
            postEventDelayed(event);
        }
    }

    public static boolean postEventDelayed(final Object event) {
        return getInstance().mHandler.post(() -> getBus().post(event));
    }

}

File: app/src/main/java/com/pixplicity/cryptogram/utils/EventProvider.kt
package com.pixplicity.cryptogram.utils

import android.os.Handler
import android.os.Looper

import com.squareup.otto.Bus

object EventProvider {

    private val handler = Handler(Looper.getMainLooper())
    val bus = Bus()

    fun postEvent(event: Any) {
        if (Looper.myLooper() == Looper.getMainLooper()) {
            bus.post(event)
        } else {
            postEventDelayed(event)
        }
    }

    fun postEventDelayed(event: Any): Boolean {
        return handler.post { bus.post(event) }
    }

    fun postEventDelayed(event: Any, delayMillis: Long): Boolean {
        return handler.postDelayed({ bus.post(event) }, delayMillis)
    }

}
